core/src/main/java/org/apache/calcite/util/TimestampWithTimeZoneString.java

```diff
@@ -202,4 +212,17 @@ public TimestampString getLocalTimestampString() {
   public TimeZone getTimeZone() {
     return timeZone;
   }
+
+  private static DateTimeUtils.PrecisionTime parseDateTime(String tsStr, TimeZone timeZone) {
+    DateTimeUtils.PrecisionTime pt =
+        DateTimeUtils.parsePrecisionDateTimeLiteral(tsStr, requireNonNull(TIMESTAMP_FORMAT.get()),
+            timeZone, -1);
+
+    if (pt == null) {
+      throw RESOURCE.illegalLiteral("TIMESTAMP WITH LOCAL TIME ZONE", tsStr,
```

Comment:
it is very useful for users to be able to see the illegal timestamp, can it be included in the message?

Reply:
It already done, i fix a tests